src/util/instanceConfigInheritance.tsx

```diff
@@ -131,12 +131,47 @@ export const figureInheritedRootStorage = (
         .filter(isDiskDevice)
         .find((device) => device.path === "/");
       if (rootDevice) {
-        return [rootDevice.pool, `${profileName} profile`];
+        return [rootDevice, `${profileName} profile`];
+      }
+    }
+  }
+
+  return [null, "LXD"];
+};
+
+export interface InheritedVolume {
+  key: string;
+  disk: LxdDiskDevice;
+  source: string;
+}
+
+export const figureInheritedVolumes = (
+  values: SharedFormTypes,
+  profiles: LxdProfile[]
+): InheritedVolume[] => {
+  const inheritedVolumes: InheritedVolume[] = [];
+  if (Object.prototype.hasOwnProperty.call(values, "profiles")) {
+    const appliedProfiles = [
+      ...(values as CreateInstanceFormValues | EditInstanceFormValues).profiles,
+    ].reverse();
+    for (const profileName of appliedProfiles) {
+      const profile = profiles.find((profile) => profile.name === profileName);
+      if (!profile) {
+        continue;
       }
+      Object.entries(profile.devices)
+        .filter(([key, network]) => isDiskDevice(network) && key !== "root")
```

Comment:
Huh? This looks a bit confusing. Maybe:
```suggestion
        .filter(([key, device]) => isDiskDevice(device) && key !== "root")
```